modules/jolt_physics/objects/jolt_soft_body_3d.cpp

```diff
@@ -424,6 +424,134 @@ bool JoltSoftBody3D::is_sleeping() const {
+
+	ERR_FAIL_NULL(shared);
+	ERR_FAIL_INDEX(p_index, (int)shared->mesh_to_physics.size());
+	const size_t physics_index = (size_t)shared->mesh_to_physics[p_index];
+
+	if (pinned_vertices.has(physics_index)) {
+		return;
+	}
+
+	const float last_step = space->get_last_step();
+	if (unlikely(last_step == 0.0f)) {
+		return;
+	}
+	wake_up();
+
+	JoltWritableBody3D body = space->write_body(jolt_id);
+	ERR_FAIL_COND(body.is_invalid());
+
+	JPH::SoftBodyMotionProperties &motion_properties = static_cast<JPH::SoftBodyMotionProperties &>(*body->GetMotionPropertiesUnchecked());
+
+	JPH::Array<JPH::SoftBodyVertex> &physics_vertices = motion_properties.GetVertices();
+	JPH::SoftBodyVertex &physics_vertex = physics_vertices[physics_index];
+
+	const JPH::Vec3 force = to_jolt(p_force);
+	physics_vertex.mForce += force;
+}
+
+void JoltSoftBody3D::apply_central_impulse(const Vector3 &p_impulse) {
+	ERR_FAIL_COND_MSG(!in_space(), vformat("Failed to apply central impulse for '%s'. Doing so without a physics space is not supported when using Jolt Physics. If this relates to a node, try adding the node to a scene tree first.", to_string()));
+
+	ERR_FAIL_NULL(shared);
+
+	const float last_step = space->get_last_step();
+	if (unlikely(last_step == 0.0f)) {
+		return;
+	}
+	wake_up();
+
+	JoltWritableBody3D body = space->write_body(jolt_id);
+	ERR_FAIL_COND(body.is_invalid());
+
+	JPH::SoftBodyMotionProperties &motion_properties = static_cast<JPH::SoftBodyMotionProperties &>(*body->GetMotionPropertiesUnchecked());
+
+	JPH::Array<JPH::SoftBodyVertex> &physics_vertices = motion_properties.GetVertices();
+
+	const int mesh_vertex_count = shared->mesh_to_physics.size();
+
+	const JPH::Vec3 impulse = to_jolt(p_impulse) / mesh_vertex_count;
+
+	for (int i = 0; i < mesh_vertex_count; ++i) {
+		const size_t physics_index = (size_t)shared->mesh_to_physics[i];
+
+		if (pinned_vertices.has(physics_index)) {
+			continue;
+		}
+
+		JPH::SoftBodyVertex &physics_vertex = physics_vertices[physics_index];
+
+		physics_vertex.mVelocity += impulse * physics_vertex.mInvMass;
+	}
+}
+
+void JoltSoftBody3D::apply_central_force(const Vector3 &p_force) {
+	ERR_FAIL_COND_MSG(!in_space(), vformat("Failed to apply central force for '%s'. Doing so without a physics space is not supported when using Jolt Physics. If this relates to a node, try adding the node to a scene tree first.", to_string()));
+
+	ERR_FAIL_NULL(shared);
+
+	const float last_step = space->get_last_step();
+	if (unlikely(last_step == 0.0f)) {
+		return;
+	}
+	wake_up();
+
+	JoltWritableBody3D body = space->write_body(jolt_id);
+	ERR_FAIL_COND(body.is_invalid());
+
+	JPH::SoftBodyMotionProperties &motion_properties = static_cast<JPH::SoftBodyMotionProperties &>(*body->GetMotionPropertiesUnchecked());
+
+	JPH::Array<JPH::SoftBodyVertex> &physics_vertices = motion_properties.GetVertices();
+
+	const int mesh_vertex_count = shared->mesh_to_physics.size();
+
+	const JPH::Vec3 force = to_jolt(p_force) / mesh_vertex_count;
```

Comment:
There is `void BodyInterface::AddForce(const BodyID &inBodyID, Vec3Arg inForce, EActivation inActivationMode)` but I'm realizing now that it has a bug because it applies the same force to all vertices (and thus applies too much force in total). I'll fix that.

Reply:
I'm going to just use that function, even tho it's broken right now.